llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -705,7 +707,25 @@ def fsdp_wrap_fn(self, module: nn.Module) -> bool:
 
     # Activation Checkpointing
     def activation_checkpointing_fn(self, module: nn.Module) -> bool:
-        return isinstance(module, MPTBlock)
+        if not hasattr(self.config, 'activation_checkpointing_target'):
+            return isinstance(module, MPTBlock)
+        act_ckpt_str = self.config.activation_checkpointing_target
+        act_ckpt_lst = act_ckpt_str.replace(' ', '').split(',')
+        if act_ckpt_lst:
+            if 'MPTBlock' in act_ckpt_lst or 'mptblock' in act_ckpt_lst:
+                act_ckpt_lst = ['MPTBlock']
```

Comment:
why cancel out all other typing?

Reply:
mptblock is layer-level, other module are sub-layer, if one checkpoints a layer, no need to checkpoint what's inside

Reply:
Why not?
It might be a bad idea (we can add a warning if you want), but it should be possible.

eg you can act ckpt the MPTBlock and also the attn_block. why not?

Reply:
act ckpt the MPTBlock meaning you recompute the entire block and do not keep the intermediate activations within the block. What does it mean if you act checkpoint attn_block on top of MPTBlock?

Reply:
a block is
-> ln -> Wqkv -> attn -> Wo -> residual_add -> ln -> W1 -> gelu -> W2 -> residual_add ->

act ckpt the block means ONLY the input to that entire block is stored durring fwd pass. Then during the bwd pass, the block does another fwd to recompute intermediate activations

If we act ckpt MPTBlock and also the attn_block, I'd expect the second fwd pass to not store any of the intermediate activations in the attn_block, and then to do a third fwd pass through the attn_block for its bwd. but maybe there this doesn't work for some reason 🤷‍♂️ 

Also image for some reason someone adds more layers (eg attn layers) outside of an MPTBlock and wants to act ckpt those...

we're talking about edge cases i guess. 
either way, either allow for bad behavior (and warn the user that they are doing something weird) or do what you're doing (but again warn the user that you are removing other modules from the act_ckpt list.